BusinessTampereTrafficMonitoring/stream_reader/stream_reader.py

```diff
@@ -77,14 +77,15 @@ def read_buffer_to_cache(self):
             t = time.time()
             try:
                 frame = self.buffer.get()
-            except queue.Empty():
-                print("Unexpected error happened, reinitializing VideoCapture")
+            except queue.Empty:
+                logger.error("Unexpected error happened, reinitializing VideoCapture")
                 self.cap.release()
                 self.cap = cv2.VideoCapture(self.video_location, apiPreference=cv2.CAP_FFMPEG)
                 while not self.cap.isOpened():
                     self.cap = cv2.VideoCapture(self.video_location, apiPreference=cv2.CAP_FFMPEG)
                     time.sleep(1)
                     if time.time() - self.t_latest_frame > 5:
+                        logger.error("Couldn't recover. More info: " + str(queue.Empty))
```

Comment:
This won't work (`str(queue.Empty)` is `"<class '_queue.Empty'>"`).

Usually you'd do something like 
```python
try:
    ...
except queue.Empty as e:
    print(f"Error: {e}")
```
to log exceptions, but in this case it seems like the queue.Empty exceptions don't contain any further information so the above would just print `"Error: "`. I also noticed that `queue.Empty` doesn't ever happen if the [`.get()` method](https://docs.python.org/3/library/queue.html#queue.Queue.get) is not given `block=False` or a timeout as a parameter.

Reply:
I had some local changes in my local separate test including arguments for get: (timeout=2, block=True). Modified them into the pull request.